src/app.service.ts

```diff
@@ -72,10 +72,10 @@ export class AppService {
     input: DocumentInput,
     options?: { merge?: boolean },
   ): Promise<Document> {
-    return await this.database.withTransaction(async (conn) => {
+    return await this.database.withTransactionAndRetry(async (conn) => {
```

Comment:
@Ken-Michalak my only worry here is that they are concurrent, so would try retry at the exact same time.

Reply:
I don't think this retry will actually do anything right now, because the `FOR UPDATE` actually prevents 2 update operations on that row.
If we run into other errors that make sense to retry like too many connections or something, we could use this logic for that by just adding to the "shouldRetry" logic.